csrc/host_ir/jit.cpp

```diff
@@ -0,0 +1,383 @@
+  }
+};
+
+// PIMPL implementation for HostIrJit
+struct HostIrJit::LlvmJitImpl {
+  std::unique_ptr<llvm::orc::LLJIT> jit;
+  std::unordered_map<const kir::Allocate*, allocate_fn> allocate_funcs_;
+  std::unique_ptr<HostIrJitParams> params_;
+  LlvmJitImpl() = default;
+  ~LlvmJitImpl() = default; // unique_ptr handles cleanup automatically
+};
+
+// Helper function to exit on error on LLVM JIT initialization
+template <typename T>
+T ExitOnErr(llvm::Expected<T>&& E) {
+  if (!E) {
+    NVF_ERROR(
+        false,
+        "LLVM JIT Initialization Error: ",
+        llvm::toString(E.takeError()));
+    exit(1);
+  }
+  return std::move(*E);
+}
+
+inline void ExitOnErr(llvm::Error&& Err) {
+  if (Err) {
+    NVF_ERROR(
+        false,
+        "LLVM JIT Initialization Error: " + llvm::toString(std::move(Err)));
+    exit(1);
+  }
+}
+
+// Helper function to create global variables for shared parameters, not used for now
+void createGlobalVariables(llvm::Module* mod, const HostIrJitParams& params) {
+  // // Note: We're now using constants instead of global variables for better performance
+  // // This function is kept for potential future use if runtime-modifiable globals are needed
+  
+  // llvm::LLVMContext& context = mod->getContext();
+  // llvm::Type* int64_type = llvm::Type::getInt64Ty(context);
+  // llvm::Type* int32_type = llvm::Type::getInt32Ty(context);
+  // llvm::Type* bool_type = llvm::Type::getInt1Ty(context);
+  // llvm::Type* ptr_type = llvm::PointerType::getUnqual(llvm::Type::getInt8Ty(context));
+  
+  // // Example: Global memory pool size (if needed for runtime configuration)
+  // if (!mod->getGlobalVariable("memory_pool_size")) {
+  //   new llvm::GlobalVariable(
+  //       *mod,
+  //       int64_type,
+  //       false,
+  //       llvm::GlobalValue::InternalLinkage,
+  //       llvm::ConstantInt::get(int64_type, 1024 * 1024 * 1024), // 1GB default
+  //       "memory_pool_size");
+  // }
+  
+  // // Example: Global stream ID (if needed for runtime configuration)
+  // if (!mod->getGlobalVariable("default_stream_id")) {
+  //   new llvm::GlobalVariable(
+  //       *mod,
+  //       int32_type,
+  //       false,
+  //       llvm::GlobalValue::InternalLinkage,
+  //       llvm::ConstantInt::get(int32_type, 0), // Default stream
+  //       "default_stream_id");
+  // }
+  
+}
+
+// Generate a function that calls at::native::empty_strided_cuda
+void generateAllocateFunc(
+    const kir::Allocate* allocate,
+    llvm::Module* mod,
+    const HostIrJitParams& params) {
+
+
+
+  at::ScalarType type = data_type_to_aten(allocate->buffer()->dtype() == DataType::Index ? PrimDataType::Int : allocate->buffer()->dtype());
+
+  llvm::LLVMContext& context = mod->getContext();
+  llvm::IRBuilder<> builder(context);
+ std::string func_name = "create_tensor_from_sizes_" +
+          std::to_string(reinterpret_cast<uintptr_t>(allocate));
+
+  // Define function signature: at::Tensor* func(int64_t*, int64_t, int64_t*, int64_t)
```

Comment:
note for myself. I'm ignoring the actual llvm stuff below. I think my review comments for this PR is already overwhelming. We can fix this as we expand the actual jit functionality later.